sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -136,6 +136,9 @@
     private RxPartitionKeyRangeCache partitionKeyRangeCache;
     private Map<String, List<PartitionKeyAndResourceTokenPair>> resourceTokensMap;
     private final boolean contentResponseOnWriteEnabled;
+    private boolean queryPlanCachingEnabled;
+    private static final int CACHE_SIZE = 30; // This value will be made configurable in future
```

Comment:
Should we really veer make this configurable? And 30 is the number of query plans to be cached? I would argue that this value should be artificially high - so that we basically nearly always are able to cache query plans. Like at least in the order of thousands. Even if query plans are relatively large - like in the order of KBs if a customer is running a workload with hundreds of different queries a couple of MBs for this case memory will be the least of their problems :-) - Disabling/reducing config as System properties - fine - but not public surface area would be my recommendation.

Reply:
With the initial data I have from some observations, I see that not many different varieties of queries are being used by the users and if parameterized, they should be good with a small size. Also, we want to start with a small size, get some feedback on the results and then either make it unbounded/very large or limit it to a particular size

Reply:
Increased it to 1000. As this is experimental feature at this point, based on the feedback and observations we make, we can set a better value